File: 144-binary-tree-preorder-traversal/144-binary-tree-preorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    List<Integer> ans = new ArrayList<>();
    public void preorderTraversalUtil(TreeNode root) {
        if (root != null) {
            ans.add(root.val);
            this.preorderTraversalUtil(root.left);
            this.preorderTraversalUtil(root.right);
        }
    }
    public List<Integer> preorderTraversal(TreeNode root) {
        this.preorderTraversalUtil(root);
        return ans;
    }
}

File: 144-binary-tree-preorder-traversal/144-binary-tree-preorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> ans;
    void preorderTraversalUtil(TreeNode* root) {
        if (root) {
            ans.push_back(root->val);
            preorderTraversalUtil(root->left);
            preorderTraversalUtil(root->right);
        }
    }
    vector<int> preorderTraversal(TreeNode* root) {
        preorderTraversalUtil(root);
        return ans;
    }
};